API/Controllers/RoleController.cs

```diff
@@ -0,0 +1,311 @@
+            if(!roles.Any())
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting all roles.",
+                    Detail = "There where no roles found in the database.",
+                    Instance = "3EB1E953-96D7-45FE-8C5C-05306AF8D060"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(mapper.Map<IEnumerable<Role>, IEnumerable<RoleResourceResult>>(roles));
+        }
+
+        /// <summary>
+        ///     Get all Scopes.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("Scopes")]
+        [Authorize(Policy = nameof(Defaults.Scopes.RoleRead))]
+        public IActionResult GetAllPossibleScopes()
+        {
+            List<string> scopeList = roleService.GetValidScopes();
+            if(!scopeList.Any())
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "No valid Scopes found",
+                    Detail = "There where no valid scopes found.",
+                    Instance = "DEB2161D-A8E7-4AAE-BB0F-CDB3CA5D5B9E"
+                };
+                return NotFound(problem);
+            }
+            return Ok(scopeList);
+        }
+
+        /// <summary>
+        ///     Get a Role.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{roleId}")]
+        [Authorize(Policy = nameof(Defaults.Scopes.RoleRead))]
+        public async Task<IActionResult> GetRole(int roleId)
+        {
+            if(roleId < 0)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting role.",
+                    Detail = "The Id is smaller then 0 and therefore it could never be a valid role id.",
+                    Instance = "5024ADDA-6DE2-4B49-896A-526E8EC4313D"
+                };
+                return BadRequest(problem);
+            }
+
+            Role role = await roleService.FindAsync(roleId);
+            if(role == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting role.",
+                    Detail = "The roule could not be found in the database.",
+                    Instance = "1739EFA6-3F31-4C88-B596-74DA403AC51B"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(mapper.Map<Role, RoleResourceResult>(role));
+        }
+
+        /// <summary>
+        /// Creates the role asynchronous.
+        /// </summary>
+        /// <param name="roleResource">The role resource.</param>
+        /// <returns></returns>
+        [HttpPost]
+        [Authorize(Policy = nameof(Defaults.Scopes.RoleWrite))]
+        public async Task<IActionResult> CreateRoleAsync([FromBody] RoleResource roleResource)
```

Comment:
Why specify FromBody here? Isn't it clear? I think APIController handles this

Reply:
The From Body tag is indeed not required since ASP.NET Core automatically takes the data from the request body unless specified otherwise.
I personally do not mind the additional FromBody tag. It makes it more clear where the data comes from, especcially for developers that do not know that it is the default.